pkg/controller/azure/provider/resourcegroup_test.go

```diff
@@ -0,0 +1,690 @@
+	fakerg "github.com/crossplaneio/crossplane/pkg/clients/azure/resourcegroup/fake"
+	"github.com/crossplaneio/crossplane/pkg/test"
+)
+
+const (
+	namespaceRG = "cool-namespace"
+	uid         = types.UID("definitely-a-uuid")
+	name        = "cool-rg"
+	location    = "coolplace"
+	// subscription = "totally-a-uuid"
+
+	providerSecretName = "cool-azure-secret"
+	providerSecretKey  = "credentials"
+	providerSecretData = "definitelyjson"
+)
+
+var (
+	errorBoom = errors.New("boom")
+
+	provider = azurev1alpha1.Provider{
+		ObjectMeta: metav1.ObjectMeta{Namespace: namespaceRG, Name: providerName},
+		Spec: azurev1alpha1.ProviderSpec{
+			Secret: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: providerSecretName},
+				Key:                  providerSecretKey,
+			},
+		},
+		Status: azurev1alpha1.ProviderStatus{
+			ConditionedStatus: corev1alpha1.ConditionedStatus{
+				Conditions: []corev1alpha1.Condition{{Type: corev1alpha1.Ready, Status: corev1.ConditionTrue}},
+			},
+		},
+	}
+
+	providerSecret = corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Namespace: namespaceRG, Name: providerSecretName},
+		Data:       map[string][]byte{providerSecretKey: []byte(providerSecretData)},
+	}
+)
+
+type resourceModifier func(*azurev1alpha1.ResourceGroup)
+
+func withConditions(c ...corev1alpha1.Condition) resourceModifier {
+	return func(r *azurev1alpha1.ResourceGroup) { r.Status.ConditionedStatus.Conditions = c }
+}
+
+func withFinalizers(f ...string) resourceModifier {
+	return func(r *azurev1alpha1.ResourceGroup) { r.ObjectMeta.Finalizers = f }
+}
+
+func withReclaimPolicy(p corev1alpha1.ReclaimPolicy) resourceModifier {
+	return func(r *azurev1alpha1.ResourceGroup) { r.Spec.ReclaimPolicy = p }
+}
+
+func withName(n string) resourceModifier {
+	return func(r *azurev1alpha1.ResourceGroup) { r.Status.Name = n }
+}
+
+func withSpecName(n string) resourceModifier {
+	return func(r *azurev1alpha1.ResourceGroup) { r.Spec.Name = n }
+}
+
+func withDeletionTimestamp(t time.Time) resourceModifier {
+	return func(r *v1alpha1.ResourceGroup) { r.ObjectMeta.DeletionTimestamp = &metav1.Time{Time: t} }
+}
+
+// func withDeletionTimestamp(t time.Time) resourceModifier {
+// 	return func(r *azurev1alpha1.ResourceGroup) { r.ObjectMeta.DeletionTimestamp = &metav1.Time{Time: t} }
+// }
+
+func resource(rm ...resourceModifier) *azurev1alpha1.ResourceGroup {
+	r := &azurev1alpha1.ResourceGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:  namespaceRG,
+			Name:       name,
+			UID:        uid,
+			Finalizers: []string{},
+		},
+		Spec: azurev1alpha1.ResourceGroupSpec{
+			Name:        name,
+			Location:    location,
+			ProviderRef: corev1.LocalObjectReference{Name: providerName},
+		},
+		Status: azurev1alpha1.ResourceGroupStatus{},
+	}
+
+	for _, m := range rm {
+		m(r)
+	}
+
+	return r
+}
+
+// Test that our ReconcilerRG implementation satisfies the ReconcilerRG interface.
```

Comment:
You mean satisfies the `reconcile.Reconciler` interface? This confused me for a second.

Reply:
Yep, I had bulk changed Reconciler to ReconcilerRG due to the conflicts with the Provider Reconciler.